HEC.FDA.ModelTest/integrationtests/UncertainStageDamageTests.cs

```diff
@@ -0,0 +1,289 @@
+﻿using HEC.FDA.Model.compute;
+using HEC.FDA.Model.hydraulics;
+using HEC.FDA.Model.hydraulics.Interfaces;
+using HEC.FDA.Model.hydraulics.Mock;
+using HEC.FDA.Model.paireddata;
+using HEC.FDA.Model.stageDamage;
+using HEC.FDA.Model.structures;
+using RasMapperLib;
+using Statistics;
+using Statistics.Distributions;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace HEC.FDA.ModelTest.integrationtests
+{
+    [Trait("RunsOn", "Remote")]
+
+    public class UncertainStageDamageTests
+    {
+        #region H&H Data
+        private static int quantityProfilesAndStructuresPerImpactArea = 4;
+
+        //First impact area
+        private static ContinuousDistribution lp3 = new LogPearson3(2.8, .438, .075, 50);
+        private static double[] RatingCurveFlows = { 0, lp3.InverseCDF(1-0.5), lp3.InverseCDF(1-0.2), lp3.InverseCDF(1-0.1), lp3.InverseCDF(1-0.04), lp3.InverseCDF(1-0.02), lp3.InverseCDF(1-0.01), lp3.InverseCDF(1-0.004), lp3.InverseCDF(1-0.002) };
+        private static IDistribution[] StageDistributions =
+        {
+            new Normal(458,0.00001),
+            new Normal(468,.312),
+            new Normal(469,.362),
+            new Normal(470,.422),
+            new Normal(471,.456),
+            new Normal(473,.474),
+            new Normal(475,.5),
+            new Normal(477,.5),
+            new Normal(479,.5)
+        };
+        private static UncertainPairedData dischargeStage = new UncertainPairedData(RatingCurveFlows, StageDistributions, new CurveMetaData("flows", "stages", "rating curve"));
+
+        //Second impact area 
+        private static double[] probabilities = new double[] { .5, .2, .1, .04, .02, .01, .004, .002 };
+        private static double[] stages = new double[] { 468, 469, 470, 471, 473, 475, 477, 479 };
+        private static int equivalentRecordLength = 50;
+        private static CurveMetaData stageFreqMetaData = new CurveMetaData("probability", "stages", "graphical stage frequency");
+        private static GraphicalUncertainPairedData frequencyStage = new GraphicalUncertainPairedData(probabilities, stages, equivalentRecordLength, stageFreqMetaData, true);
+
+        //Hydraulics for a given impact area 
+        private static HydraulicDataset ComputeHydraulicDataset()
+        {
+            List<IHydraulicProfile> dummyHydraulicProfiles = new List<IHydraulicProfile>();
+            for (int i = 0; i < stages.Length; i++)
+            {
+                float[] profileStages = new float[quantityProfilesAndStructuresPerImpactArea];
```

Comment:
I dont like this name. maybe make two static ints. ProfilesPerImpactArea and StructuresPerImpactArea. Those two things don't inherently match and I, finding it confusing reading it.

Reply:
the two things have to match because they're tied together by index